src/services/findAllReferences.ts

```diff
@@ -1204,8 +1204,13 @@ namespace ts.FindAllReferences {
                     return undefined;
                 }
 
-                // The search scope is the container node
-                scope = container;
+                if (isFunctionExpression(container)) {
+                    scope = findAncestor(container.parent, n =>
+                        isBinaryExpression(n) || isVariableDeclaration(n) || isVariableDeclarationList(n) ? false : isStatement(n) || "quit") || container;
+                }
+                else {
+                    scope = container;
+                }

```

Comment:
It would be better to use the existing function for finding jsdoc locations:

```suggestion
                scope = container;
                if (isFunctionExpression(container)) {
                    let next;
                    while (next = getNextJSDocCommentLocation(scope)) {
                        scope = next;
                    }
                }
```

although you'll have to go in and export it from `src/compiler/utilities.ts`

Reply:
@sandersn I didn't about this util function :). Thanks.